lib/CSharp2ClientGenerator.php

```diff
@@ -559,7 +628,10 @@ function writeClass(DOMElement $classNode)
 						$this->appendLine("				this._$dotNetPropName = ParseBool(node[\"$propName\"].Value<string>());");
 						break;
 					case "float":
-						$this->appendLine("				this._$dotNetPropName = ParseFloat(node[\"$propName\"].Value<string>());");
+						$this->appendLine("				this._$dotNetPropName = ParseFloat(node[\"$propName\"].Value<string>());");						
+						
+						//BEO-9746
+						$this->appendLine("				this._{$dotNetPropName}AsDouble  = ParseDouble(node[\"$propName\"].Value<string>());");

```

Comment:
Thast a bit of a probelm.. 
the setter of the float value is setting both the float and asDouble variant using cast .. (which may losses precision)

I would do something different in the CTOR .. 

i would store the value in a private field with `double` type .. 
and will provide getters \setter to that private in both float (which will cast the private field) and the double property whoch will return or set the value as-is ..